### DISCUSSION TITLE
How can I re-run the Alpine x-for loop, after a change in the Alpine variable (due to Livewire update)

### DISCUSSION THREAD

**Discussion by joffreypersia** [livewire/livewire#7448] [1. Help]
Hi, 

I am building a Livewire component that is a mosaic of images with a state "processed" or "not processed" with a little icon of loading if the image is processing. 

I use a template x-for to loop through my medias. 
After 2 sec, one media is processed, and the new JSON that is passed to the component is changed. 
I see the init() working again because of my console.log(this.medias) is triggered again, but the loading state media.checked is not changed inside the Alpine loop. 

What do you think? Is there a way to force Alpine to re-run the loop? 

Thank you,
Joffrey

**Comment by joffreypersia** [comment]
OK, I figured out that it is caused by Alpine not recognizing the changes. If I entangle the array items and modify it in the class itself, the loop is re-rendered. 

So, I should not use json if the data is dynamic.

**Comment by duemti** [comment]
A clever solution is this:

```
x-data="{
    renderKey: 0,

    init() {
        this.renderKey++; // changing this will force re-run of x-for
    }
}"

<template x-for="(field, key) in data" :key="key + '_' + renderKey">
</template>
```
